test/test/basic-channel.js

```diff
@@ -0,0 +1,96 @@
+      const beforeSubBalanceAlice = await subClient.queryAccountBalance(polkadotSenderSS58Alice);
+      const beforeSubBalanceBob = await subClient.queryAccountBalance(polkadotSenderSS58Bob);
+      const beforeSubBalanceCharlie = await subClient.queryAccountBalance(polkadotSenderSS58Charlie);
+      const beforeSubBalanceDave = await subClient.queryAccountBalance(polkadotSenderSS58Dave);
+      const beforeSubBalanceEve = await subClient.queryAccountBalance(polkadotSenderSS58Eve);
+      const beforeSubBalanceFerdie = await subClient.queryAccountBalance(polkadotSenderSS58Ferdie);
+
+      // lock DOT using basic channel
+      await subClient.api.tx.sudo.sudo(
+        subClient.api.tx.utility.batchAll([
+          subClient.api.tx.utility.dispatchAs(
+            {system: {signed: subClient.alice.address}},
+            subClient.api.tx.dotApp.lock(ChannelId.BASIC, ethAccount, amount.toFixed())
+          ),
+          subClient.api.tx.utility.dispatchAs(
+            {system: {signed: subClient.bob.address}},
+            subClient.api.tx.dotApp.lock(ChannelId.BASIC, ethAccount, amount.toFixed())
+          ),
+          subClient.api.tx.utility.dispatchAs(
+            {system: {signed: subClient.charlie.address}},
+            subClient.api.tx.dotApp.lock(ChannelId.BASIC, ethAccount, amount.toFixed())
+          ),
+          subClient.api.tx.utility.dispatchAs(
+            {system: {signed: subClient.dave.address}},
+            subClient.api.tx.dotApp.lock(ChannelId.BASIC, ethAccount, amount.toFixed())
+          ),
+          subClient.api.tx.utility.dispatchAs(
+            {system: {signed: subClient.eve.address}},
+            subClient.api.tx.dotApp.lock(ChannelId.BASIC, ethAccount, amount.toFixed())
+          ),
+          subClient.api.tx.utility.dispatchAs(
+            {system: {signed: subClient.ferdie.address}},
+            subClient.api.tx.dotApp.lock(ChannelId.BASIC, ethAccount, amount.toFixed())
+          ),
+        ])
+      ).signAndSend(sudoPair)
+      await ethClient.waitForNextEventData({ appName: 'snowDOT', eventName: 'Minted' });
+
+      const afterEthBalance = await ethClient.getDotBalance(ethAccount);
+      const afterSubBalanceAlice = await subClient.queryAccountBalance(polkadotSenderSS58Alice);
+      const afterSubBalanceBob = await subClient.queryAccountBalance(polkadotSenderSS58Bob);
+      const afterSubBalanceCharlie = await subClient.queryAccountBalance(polkadotSenderSS58Charlie);
+      const afterSubBalanceDave = await subClient.queryAccountBalance(polkadotSenderSS58Dave);
+      const afterSubBalanceEve = await subClient.queryAccountBalance(polkadotSenderSS58Eve);
+      const afterSubBalanceFerdie = await subClient.queryAccountBalance(polkadotSenderSS58Ferdie);
+
+      expect(afterEthBalance.minus(beforeEthBalance)).to.be.bignumber.equal(6 * amountWrapped);
+      expect(beforeSubBalanceAlice.minus(afterSubBalanceAlice)).to.be.bignumber.gte(amount);
```

Comment:
This was a bit painful: turns out that chai has greaterThanOrEqual, but that failed here because it expected a number or a date. The [chai-bignumber plugin](https://www.chaijs.com/plugins/chai-bignumber/) we're using here has *almost* all comparison assertions, except lessThanOrEqual and greaterThanOrEqual. Those are still the base chai assertions, which fail when used on BigNumbers.

Reply:
I'm not sure I understand here. This was more about incompleteness in the chai-bignumber plugin than the way we're testing. What pain points have you seen with using dotApp & co to test?

Reply:
Sorry for not being clear, it is not directly related. I have run into this issue before where I have to convert down to `Number` or `String` to assert using chai.

My comment was more general. Channels are a low level concept in the bridge. Where the various apps (e.g. dotApp) are higher level. And we are using the high level apps in this test case to test features of the lower level channel. And it would be nice to have a way to test channels directly without high level apps. Maybe instead of asserting balance updates to test the permission-less channel feature we could assert against the existence of events that the channels emit.

Also our tests which use balances for assertions are side effecting and running them can affect the outcome of other tests (e.g Depleting balances). This is fine for the rest of our tests because they are testing the high level apps directly as well as the underlying channel. I think this is our first test which is isolated to testing a feature of a channel.

But this is non blocking and is more of an observation.

Reply:
Made a PR to add the aliases [here](https://github.com/asmarques/chai-bignumber/pull/15).

Chatted a bit offline, and testing the channel separately would be ideal but we'd have to have an extrinsic to send messages across the channel. We don't want to expose `submit` directly, so a `TestChannelApp` would be best. But that's a fair bit of work for limited gain, given how simple `DOTApp` is.

@vgeddes @claravanstaden Do you think it's worth isolating this test to the channel? Using `DOTApp` seems good enough to me, but I might be missing something.